### PR TITLE
First pass at generic grid functions.

### PR DESCRIPTION
**Pull request by julesfowler**

### CODE REVIEW

**Comment by julesfowler** [comment]
So -- a better outline of what needs to happen for this PR : 
[x] Draft up the function to pull models and rescale them.
- The draft is done and working as expected but I don't think the parameter space is complete. @hrwakeford may be able to shed some light.
- Also, right now I have no clue what these parameters are, I would love to make them more verbose.
[x] Draft up the functions to display the model and save on click.
- Shameless copy of the Fortney grid.
[ ] Build the template for the page(s). 
[ ] Figure out how we want to incorporate the database. Throw it in the repo or? 

[ ] BONUS : Have some option to see the mode from which it is scaled.

**Comment by julesfowler** [comment]
@hover2pi Pending how we want to handle the database I think this is ready for review.

**Comment by hover2pi** [comment]
I put this in Slack but will repeat here for posterity.

Right now, the Fortney and Generic model grid tools are in the Web app (as separate scripts) but not executable from a Python session. It might make more sense to make a `forward_models` subpackage that has only the generic grid code, which runs all the forward modeling and gets called by the Web app to generate the plots.

What do you think?

**Comment by julesfowler** [comment]
Yeah I was tossing this idea around myself because these are kind of unweildy to keep in the app -- especially because the Fortney grid stuff could be a lot more readable if it were broken down. 

May I ask for a compromise? 
Step 1. - [ ]  Get the Goyal Grid working. 
Step 2. - [ ] Separate it and the Fortney Grid into a `forward_models` subpackage?

**Comment by hover2pi** [comment]
@julesfowler That's fine with me! So I should go ahead and review this PR?

**Comment by julesfowler** [comment]
@hover2pi Actually, probably hold off. It sounds like Hannah and I will need to make some adjustments.

**Comment by hover2pi** [comment]
@julesfowler Is this still WIP of should I review?

**Comment by julesfowler** [comment]
@hover2pi The only thing left to do is sort where that data is going. So -- if you want to review it (and ignore the fact that the database call is hardcoded to a file rn) by all means!!

File: exoctk/exoctk_app/app_exoctk.py [hunk 1]
```diff
@@ -672,6 +673,209 @@ def save_fortney_result():
...
+            
+        radius_ratio = np.sqrt(model_spectra) * inputs['r_planet']/inputs['r_star']
+        r_star = inputs['r_star'] * r_sun
+        r_planet = inputs['r_planet'] * r_jupiter
+        model_grav = model_grav * 1e2
+        inputs['gravity'] = inputs['gravity'] * 1e2
+        
+        # Start with baseline based on model parameters
+        scale_height = (boltzmann * model_temp) / (permitivity * model_grav)
+        print(scale_height)
```
Comment (hover2pi) [inline exoctk/exoctk_app/app_exoctk.py]: Are these print statements for testing or output for the user? If for testing, nuke 'em. If for the user, perhaps add some string formatting like "Scale height: {}" or something?

File: exoctk/exoctk_app/app_exoctk.py [hunk 2]
```diff
@@ -672,6 +673,209 @@ def save_fortney_result():
...
+        # Define constants
+        boltzmann = 1.380658E-16 # gm*cm^2/s^2 * Kelvin
+        permitivity = 1.6726E-24 * 2.3 #g  cgs  Hydrogen + Helium Atmosphere
+        optical_depth = 0.56 
+        r_sun = 69580000000 # cm
+        r_jupiter = 6991100000 # cm
+ 
+        closest_match = {'model_key': model_key, 'model_gravity': model_grav,
+                         'model_temperature': model_temp}
+        generic_db = '/user/jfowler/exoctk_work/generic/generic_grid_db.hdf5'
```
Comment (hover2pi) [inline exoctk/exoctk_app/app_exoctk.py]: Replace with environment variable?

File: exoctk/exoctk_app/app_exoctk.py [hunk 3]
```diff
@@ -672,6 +673,209 @@ def save_fortney_result():
...
+        sort = np.argsort(model_wv)
+        solution['wv'] = model_wv[sort]
+        solution['radius'] = solution['radius'][sort]
+        solution['spectra'] = (solution['radius']/r_star)**2
+    
+    except (KeyError, ValueError) as e:
+        error_message = 'One of the parameters to make up the model was missing or out of range.'
+        model_key = 'rainout_0400_50_+0.0_0.70_0010_1.00'
+        solution = {}
+        with h5py.File('/user/jfowler/exoctk_work/generic/generic_grid_db.hdf5') as f:
```
Comment (hover2pi) [inline exoctk/exoctk_app/app_exoctk.py]: Replace with environment variable

**Comment by julesfowler** [comment]
@hover2pi, I took away print statements on your suggestion, and went ahead and folded in the new data organization stuff. 

The idea is that we'll have an available zip download that puts everything in one directory, with sub directories for each data piece. One environment variable only. 

I think I updated all of the web functions, notebook examples, and things that explicitly call to those paths, but please tell me if I missed anything?

**Comment by hover2pi** [comment]
@julesfowler Oh, wait. The tests won't pass without the environment variable. Could you add something to the affected tools like `if EXOCTK_DATA is None: print("Sorry, no data could be found. Please set your environment variable.")` so the tests pass?

**Comment by julesfowler** [comment]
Look at these tests passing. 🎉 🎉 🎉 

@hover2pi I think this is ready to go in -- but we gotta make sure our data structure on the server matches it. (And if we have a reference in our installation, that too.)

**Comment by julesfowler** [comment]
@hover2pi I found a strange bug -- wherein the model grid was missing a model for 500. 
I don't have time now to see if that's a problem with how I built the database or not, but now it won't be tripping up the tool. 

Other than that -- I test repackaging the data and running the tool and it all looks good on the test sit e: https://tlexoctk1.stsci.edu/generic?temperature=2400&gravity=15&r_planet=2&r_star=2&condensation=local&metallicity=%2B0.0&c_o=0.35&haze=0001&cloud=0.00

Tell me if you think this is ready?

**Comment by julesfowler** [comment]
And now you can actually download the data in a zipped package too.
Again -- live on the website. Man is it slow -- but it seems to be working.

**Pull request merged**